### PR TITLE
Initial cleaned vehicle positions/locations model

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

I am _not_ actually trying to clean the "trips" more than right now. I want/need some input on the scope of this model. 

Example query showing too-fast position changes
```
with joined as (
    select locations.dt
         , locations.key
         , next_locations.key
         , locations.location_timestamp
         , locations.header_timestamp
         , locations.vehicle_timestamp
         , locations.position_latitude
         , locations.position_longitude
         , next_locations.location_timestamp
         , next_locations.header_timestamp
         , next_locations.vehicle_timestamp
         , next_locations.position_latitude
         , next_locations.position_longitude
         , ST_DISTANCE(locations.position, next_locations.position)                                AS DISTANCE
         , TIMESTAMP_DIFF(next_locations.location_timestamp, locations.location_timestamp, SECOND) AS seconds_in_between
         , ST_DISTANCE(locations.position, next_locations.position) /
           TIMESTAMP_DIFF(next_locations.location_timestamp, locations.location_timestamp, SECOND) AS meters_per_second
    from `cal-itp-data-infra-staging`.andrew_mart_gtfs.fct_vehicle_locations locations
             inner join `cal-itp-data-infra-staging`.andrew_mart_gtfs.fct_vehicle_locations next_locations
                        on locations.next_location_key = next_locations.key
    where locations.dt = '2022-10-26'
      and next_locations.dt = '2022-10-26'
)
select count(0) over ()
     , min(meters_per_second) over ()
     , percentile_cont(meters_per_second, 0.5) OVER ()
     , percentile_cont(meters_per_second, 0.90) OVER ()
     , percentile_cont(meters_per_second, 0.99) OVER ()
     , percentile_cont(meters_per_second, 0.999) OVER ()
     , max(meters_per_second) over ()
from joined
-- ORDER BY meters_per_second desc nulls last
;

```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

### CODE REVIEW

File: warehouse/models/mart/gtfs/_mart_gtfs.yml [hunk 1]
```diff
@@ -95,6 +95,29 @@ models:
...
+      level.
+    columns:
+      - name: key
+        description: |
+          Synthetic primary key.
+        tests:
+          - unique:
+              # TODO: make an RT test macro that does this date subtraction
+              # test hour = 0 UTC because that's 5pm Pacific = PM peak, good sample of data
+              config:
```
Comment (lauriemerrell) [inline warehouse/models/mart/gtfs/_mart_gtfs.yml:111]: do you need `config`? I didn't in mine to add a `where` to the uniqueness test

File: warehouse/models/mart/gtfs/fct_vehicle_locations.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        , COALESCE(vehicle_timestamp, header_timestamp) AS location_timestamp
+    FROM fct_vehicle_positions_messages
+    WHERE trip_id IS NOT NULL
+        AND _gtfs_dataset_name != 'Bay Area 511 Regional VehiclePositions'
+)
+
+, deduped AS (
+    SELECT *,
+        {{ dbt_utils.surrogate_key(['dt', 'base64_url', 'location_timestamp', 'vehicle_id', 'vehicle_label', 'trip_id']) }} AS key,
+        {{ dbt_utils.surrogate_key(['dt', 'base64_url', 'vehicle_id', 'vehicle_label', 'trip_id']) }} AS path_key
```
Comment (lauriemerrell) [inline warehouse/models/mart/gtfs/fct_vehicle_locations.sql:25]: I do think we might want to call this like `vehicle_trip_key` or something like that, because `pathways` is another unrelated GTFS file

File: warehouse/models/mart/gtfs/fct_vehicle_positions_messages.sql [hunk 3]
```diff
@@ -29,7 +30,7 @@ fct_vehicle_positions_messages AS (
         base64_url,
         _extract_ts,
         _config_extract_ts,
-        _name,
+        _gtfs_dataset_name,
```
Comment (lauriemerrell) [inline warehouse/models/mart/gtfs/fct_vehicle_positions_messages.sql:33]: does this need to be changed in the YAML too?

**Review by lauriemerrell — COMMENTED** [review]
generally looks good, just a few small things

**Comment by lauriemerrell** [comment]
Also just want to mention that this is one of the items mentioned in #1761

File: warehouse/models/mart/gtfs/_mart_gtfs.yml (hunk 1 above)
Comment (atvaccaro) [reply warehouse/models/mart/gtfs/_mart_gtfs.yml:111]: I think `config:` is the new thing going forward but top-level `where:` is backwards-compatible?

The docs do seem aligned on `config:` https://docs.getdbt.com/reference/test-configs#test-specific-configurations

File: warehouse/models/mart/gtfs/fct_vehicle_locations.sql (hunk 2 above)
Comment (atvaccaro) [reply warehouse/models/mart/gtfs/fct_vehicle_locations.sql:25]: 🤦 fixing

File: warehouse/models/mart/gtfs/fct_vehicle_positions_messages.sql (hunk 3 above)
Comment (atvaccaro) [reply warehouse/models/mart/gtfs/fct_vehicle_positions_messages.sql:33]: yep ty

File: warehouse/models/mart/gtfs/fct_vehicle_locations.sql [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    FROM coalesced_and_filtered
+    QUALIFY ROW_NUMBER() OVER (
+        -- the dt is necessary to preserve partition elimination in downstream queries
+        PARTITION BY dt, base64_url, location_timestamp, vehicle_id, vehicle_label, trip_id
+        ORDER BY NULL
+    ) = 1
+)
+
+, fct_vehicle_locations AS (
+    SELECT *,
```
Comment (lauriemerrell) [inline warehouse/models/mart/gtfs/fct_vehicle_locations.sql:35]: sorry for not catching before but -- I thought we try to avoid `select *` in the final CTE (for explicitness)?

**Review by lauriemerrell — APPROVED** [review]

File: warehouse/models/mart/gtfs/fct_vehicle_locations.sql (hunk 4 above)
Comment (atvaccaro) [reply warehouse/models/mart/gtfs/fct_vehicle_locations.sql:35]: I think generally yes but in this case I think it's fine because this is essentially just a subset of `fct_vehicle_position_messages`?
